homeassistant/components/netgear_lte/__init__.py

```diff
@@ -247,23 +255,68 @@ async def cleanup(event):
         """Clean up resources."""
         update_unsub()
```

Comment:
We need to stop the track time interval listener on config entry unload too, not on on Home Assistant stop.

Reply:
I thought that `entry.async_on_unload` does that. Perhaps I misunderstand what it does?

Reply:
The only place we currently use `ConfigEntry.async_on_unload` is to remove the event listener for Home Assistant stop event.

`ConfigEntry.async_on_unload` is used to call a function, the parameter passed, when the config entry unloads.